# **About Dataset:**
 This dataset contains information regarding Dunkin' stores across the US. Variables include location (in address and lat/long, store phone, store operating hours, and store features (drive-thru, mobile order, etc.).

In [1]:
import pandas as pd
from google.colab import files
uploaded = files.upload()
df = pd.read_csv("dunkin_stores.csv")

Saving dunkin_stores.csv to dunkin_stores.csv


# **Exploring Dataset:**

In [3]:
df.head()

row_id        address_line_1 address_line_2         city state    zip  \
0       0         1114 1st St N      Suite 200    Alabaster    AL  35007   
1       1   8171 US Highway 431            NaN  Albertville    AL  35950   
2       2  1700 US Highway 72 E            NaN       Athens    AL  35611   
3       3      171 N College St            NaN       Auburn    AL  36830   
4       4     2049 S College St            NaN       Auburn    AL  36832   

            phone            mon_hrs            tue_hrs            wed_hrs  \
0  (205) 624-4880  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
1  (256) 869-6075  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   
2  (256) 960-9302  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
3  (334) 707-8552  6:00 AM - 3:00 PM  6:00 AM - 3:00 PM  6:00 AM - 3:00 PM   
4  (334) 501-2233  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   

   ...             sat_hrs            sun_hrs drive-thru mobile-order  \
0  ...   6:00 AM - 8:00 PM  6:00 AM - 8:00 PM       True        False   
1  ...  5:00 AM - 10:00 PM  5:00 AM - 9:00 PM       True         True   
2  ...   5:00 AM - 8:00 PM  5:00 AM - 8:00 PM       True         True   
3  ...   6:00 AM - 3:00 PM  6:00 AM - 3:00 PM      False        False   
4  ...   5:00 AM - 8:00 PM  6:00 AM - 8:00 PM       True         True   

   dunkin-card   kcup  curbside-pickup  has-baskin-robbins    loc_lat  \
0         True   True            False               False        NaN   
1         True  False            False                True        NaN   
2         True   True            False               False  34.781698   
3         True   True            False               False  32.608800   
4         True   True            False               False  32.567323   

    loc_long  
0        NaN  
1        NaN  
2 -86.934524  
3 -85.481849  
4 -85.501589  

[5 rows x 22 columns]

In [4]:
df.tail()


row_id             address_line_1 address_line_2              city  \
9243    9243     106 W Silver Spring Dr          Ste A     Whitefish Bay   
9244    9244  2415 Wisconsin Dells Pkwy            NaN   Wisconsin Dells   
9245    9245             69 Commerce St            NaN   Wisconsin Dells   
9246    9246              2121 8th St S            NaN  Wisconsin Rapids   
9247    9247            6305 Missile Dr            NaN     FE Warren AFB   

     state    zip           phone            mon_hrs            tue_hrs  \
9243    WI  53217  (414) 284-7608  6:00 AM - 6:00 PM  6:00 AM - 6:00 PM   
9244    WI  53965  (608) 253-5353  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   
9245    WI  53965  (608) 350-9029  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   
9246    WI  54494  (715) 204-5260  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
9247    WY  82005  (307) 637-2818  6:00 AM - 5:00 PM  6:00 AM - 5:00 PM   

                wed_hrs  ...            sat_hrs            sun_hrs drive-thru  \
9243  6:00 AM - 6:00 PM  ...  6:00 AM - 6:00 PM  6:00 AM - 4:00 PM      False   
9244  5:30 AM - 8:00 PM  ...  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM       True   
9245  5:30 AM - 8:00 PM  ...  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM       True   
9246  5:00 AM - 8:00 PM  ...  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM       True   
9247  6:00 AM - 5:00 PM  ...  8:00 AM - 3:00 PM  8:00 AM - 3:00 PM       True   

     mobile-order  dunkin-card  kcup  curbside-pickup  has-baskin-robbins  \
9243        False         True  True            False               False   
9244         True         True  True            False               False   
9245         True         True  True            False               False   
9246        False         True  True            False               False   
9247         True         True  True            False                True   

        loc_lat   loc_long  
9243        NaN        NaN  
9244  43.625933 -89.787100  
9245        NaN        NaN  
9246  44.375934 -89.816898  
9247        NaN        NaN  

[5 rows x 22 columns]

In [5]:
df.shape

(9248, 22)

In [6]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9248 entries, 0 to 9247
Data columns (total 22 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   row_id              9248 non-null   int64  
 1   address_line_1      9248 non-null   object 
 2   address_line_2      1763 non-null   object 
 3   city                9248 non-null   object 
 4   state               9248 non-null   object 
 5   zip                 9248 non-null   int64  
 6   phone               9126 non-null   object 
 7   mon_hrs             9248 non-null   object 
 8   tue_hrs             9248 non-null   object 
 9   wed_hrs             9248 non-null   object 
 10  thu_hrs             9248 non-null   object 
 11  fri_hrs             9248 non-null   object 
 12  sat_hrs             9248 non-null   object 
 13  sun_hrs             9248 non-null   object 
 14  drive-thru          9248 non-null   bool   
 15  mobile-order        9248 non-null   bool   
 16  dunkin

In [7]:
df.nunique()

row_id                9248
address_line_1        9144
address_line_2         970
city                  3126
state                   45
zip                   5062
phone                 8989
mon_hrs                235
tue_hrs                230
wed_hrs                227
thu_hrs                230
fri_hrs                231
sat_hrs                229
sun_hrs                229
drive-thru               2
mobile-order             2
dunkin-card              2
kcup                     2
curbside-pickup          2
has-baskin-robbins       2
loc_lat               6095
loc_long              6095
dtype: int64

From this readout we can see that while there are 9248 stores located in the dataset, there are only 9144 unique addresses listed. This suggests that some addresses have more than one Dunkin' on site. These are likely large public areas such as airports or stadiums. To further investigate, we can look at the addresses that appear more than once in the set.

In [8]:
df.address_line_1.value_counts().head(20)

JFK Intl Airport                  14
Logan Airport                      8
Newark Int'L Airport               6
Ronald Reagan Airport              6
Dfw Int. Airport                   5
Philadelphia Int. Airport          4
2 Penn Plz                         3
1 Lincoln Financial Field Way      3
Wichita Airport                    3
339 Main St                        3
2 Main St                          3
4 Yawkey Way                       3
Hartsfield Jackson Atl Airport     3
Dallas Love Field Airport          3
75 Main St                         3
Dulles Intl Airport                3
Bwi Airport                        3
195 Main St                        2
Kansas Turnpike                    2
625 8th Ave                        2
Name: address_line_1, dtype: int64

As expected the list consists of airports and other large venues like sports stadiums. For example, 1 Lincoln Financial Field Way is the address of Lincoln Financial Field,which is an American football stadium located in Philadelphia, Pennsylvania,United States.



In [9]:
df.corr()

row_id       zip  drive-thru  mobile-order  dunkin-card  \
row_id              1.000000 -0.210250   -0.082110     -0.007624    -0.024763   
zip                -0.210250  1.000000    0.131469     -0.060957    -0.142236   
drive-thru         -0.082110  0.131469    1.000000      0.219475     0.303790   
mobile-order       -0.007624 -0.060957    0.219475      1.000000     0.545578   
dunkin-card        -0.024763 -0.142236    0.303790      0.545578     1.000000   
kcup                0.001966 -0.120017    0.248717      0.346439     0.575472   
curbside-pickup     0.003095 -0.024147   -0.165453      0.059771     0.113637   
has-baskin-robbins -0.000629  0.165096    0.005350      0.066413     0.093784   
loc_lat             0.267636 -0.373275   -0.078854      0.047004     0.081163   
loc_long            0.238835 -0.934167   -0.110184      0.060926     0.165668   

                        kcup  curbside-pickup  has-baskin-robbins   loc_lat  \
row_id              0.001966         0.003095           -0.000629  0.267636   
zip                -0.120017        -0.024147            0.165096 -0.373275   
drive-thru          0.248717        -0.165453            0.005350 -0.078854   
mobile-order        0.346439         0.059771            0.066413  0.047004   
dunkin-card         0.575472         0.113637            0.093784  0.081163   
kcup                1.000000         0.105222            0.067677  0.011256   
curbside-pickup     0.105222         1.000000            0.083450 -0.046647   
has-baskin-robbins  0.067677         0.083450            1.000000 -0.059153   
loc_lat             0.011256        -0.046647           -0.059153  1.000000   
loc_long            0.128884         0.019471           -0.123216  0.388220   

                    loc_long  
row_id              0.238835  
zip                -0.934167  
drive-thru         -0.110184  
mobile-order        0.060926  
dunkin-card         0.165668  
kcup                0.128884  
curbside-pickup     0.019471  
has-baskin-robbins -0.123216  
loc_lat             0.388220  
loc_long            1.000000

In [10]:
df.columns

Index(['row_id', 'address_line_1', 'address_line_2', 'city', 'state', 'zip',
       'phone', 'mon_hrs', 'tue_hrs', 'wed_hrs', 'thu_hrs', 'fri_hrs',
       'sat_hrs', 'sun_hrs', 'drive-thru', 'mobile-order', 'dunkin-card',
       'kcup', 'curbside-pickup', 'has-baskin-robbins', 'loc_lat', 'loc_long'],
      dtype='object')

# **Cleaning:**

In [11]:
df_new = df[['row_id','city', 'state','address_line_1', 'address_line_2',
       'phone', 'mon_hrs', 'tue_hrs', 'wed_hrs', 'thu_hrs', 'fri_hrs',
       'sat_hrs', 'sun_hrs', 'drive-thru', 'mobile-order', 'dunkin-card',
       'kcup', 'curbside-pickup', 'has-baskin-robbins']]
df_new.head(3)

row_id         city state        address_line_1 address_line_2  \
0       0    Alabaster    AL         1114 1st St N      Suite 200   
1       1  Albertville    AL   8171 US Highway 431            NaN   
2       2       Athens    AL  1700 US Highway 72 E            NaN   

            phone            mon_hrs            tue_hrs            wed_hrs  \
0  (205) 624-4880  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
1  (256) 869-6075  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   
2  (256) 960-9302  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   

             thu_hrs             fri_hrs             sat_hrs  \
0  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   6:00 AM - 8:00 PM   
1  5:00 AM - 9:00 PM  5:00 AM - 10:00 PM  5:00 AM - 10:00 PM   
2  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   

             sun_hrs  drive-thru  mobile-order  dunkin-card   kcup  \
0  6:00 AM - 8:00 PM        True         False         True   True   
1  5:00 AM - 9:00 PM        True          True         True  False   
2  5:00 AM - 8:00 PM        True          True         True   True   

   curbside-pickup  has-baskin-robbins  
0            False               False  
1            False                True  
2            False               False

In [12]:
df_new.isna().sum()

row_id                   0
city                     0
state                    0
address_line_1           0
address_line_2        7485
phone                  122
mon_hrs                  0
tue_hrs                  0
wed_hrs                  0
thu_hrs                  0
fri_hrs                  0
sat_hrs                  0
sun_hrs                  0
drive-thru               0
mobile-order             0
dunkin-card              0
kcup                     0
curbside-pickup          0
has-baskin-robbins       0
dtype: int64

In [13]:
df_new[df_new['phone'].isna()]

row_id            city state         address_line_1 address_line_2  \
23        23          Helena    AL         2536 Helena Rd          Ste D   
25        25          Hoover    AL            5413 US 280        Ste 100   
54        54      False Pass    AK          1700 Umiak Dr            NaN   
55        55      False Pass    AK          1800 Umiak Dr            NaN   
71        71       Flagstaff    AZ   4419 N US Highway 89            NaN   
...      ...             ...   ...                    ...            ...   
8769    8769        La Porte    TX     1900 Highway 146 S            NaN   
8785    8785   Missouri City    TX         6231 Highway 6            NaN   
8976    8976  Fredericksburg    VA          5459 Plank Rd            NaN   
9131    9131      Washington    DC  2420 New York Ave, NE            NaN   
9151    9151  Falling Waters    WV    88 Corporate Blvd W          Ste C   

     phone              mon_hrs            tue_hrs              wed_hrs  \
23     NaN    5:00 AM - 8:00 PM  5:00 AM - 8:00 PM    5:00 AM - 8:00 PM   
25     NaN    5:00 AM - 8:00 PM  5:00 AM - 8:00 PM    5:00 AM - 8:00 PM   
54     NaN    3:00 AM - 7:00 PM  3:00 AM - 7:00 PM    3:00 AM - 7:00 PM   
55     NaN  12:00 AM - 12:00 AM  3:00 AM - 7:00 PM  12:00 AM - 12:00 AM   
71     NaN    5:00 AM - 7:00 PM  5:00 AM - 7:00 PM    5:00 AM - 7:00 PM   
...    ...                  ...                ...                  ...   
8769   NaN    4:00 AM - 5:00 PM  4:00 AM - 5:00 PM    4:00 AM - 5:00 PM   
8785   NaN    5:00 AM - 9:00 PM  5:00 AM - 9:00 PM    5:00 AM - 9:00 PM   
8976   NaN    5:00 AM - 9:00 PM  5:00 AM - 9:00 PM    5:00 AM - 9:00 PM   
9131   NaN    6:00 AM - 9:00 PM  6:00 AM - 9:00 PM    6:00 AM - 9:00 PM   
9151   NaN    5:00 AM - 9:00 PM  5:00 AM - 9:00 PM    5:00 AM - 9:00 PM   

                thu_hrs            fri_hrs             sat_hrs  \
23    5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  6:00 AM - 12:00 AM   
25    5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  6:00 AM - 12:00 AM   
54    3:00 AM - 7:00 PM  3:00 AM - 7:00 PM   3:00 AM - 7:00 PM   
55        Open 24 Hours      Open 24 Hours       Open 24 Hours   
71    5:00 AM - 7:00 PM  5:00 AM - 7:00 PM   5:00 AM - 7:00 PM   
...                 ...                ...                 ...   
8769  4:00 AM - 5:00 PM  4:00 AM - 5:00 PM   5:00 AM - 5:00 PM   
8785  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   5:00 AM - 9:00 PM   
8976  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   5:00 AM - 9:00 PM   
9131  6:00 AM - 9:00 PM  6:00 AM - 9:00 PM   6:00 AM - 9:00 PM   
9151  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   5:00 AM - 9:00 PM   

                sun_hrs  drive-thru  mobile-order  dunkin-card   kcup  \
23    6:00 AM - 8:00 PM       False         False        False  False   
25    6:00 AM - 8:00 PM       False         False        False  False   
54    3:00 AM - 7:00 PM       False         False        False  False   
55        Open 24 Hours       False         False        False  False   
71    5:00 AM - 7:00 PM       False         False        False  False   
...                 ...         ...           ...          ...    ...   
8769  6:00 AM - 5:00 PM       False          True         True  False   
8785  5:00 AM - 9:00 PM        True         False         True  False   
8976  5:00 AM - 8:30 PM        True         False         True   True   
9131  6:00 AM - 9:00 PM        True          True         True   True   
9151  5:00 AM - 9:00 PM       False         False        False  False   

      curbside-pickup  has-baskin-robbins  
23              False               False  
25              False               False  
54              False               False  
55              False               False  
71              False               False  
...               ...                 ...  
8769            False               False  
8785            False               False  
8976             True               False  
9131            False                True  
9151            False               Fa

In [14]:
df_new.dtypes

row_id                 int64
city                  object
state                 object
address_line_1        object
address_line_2        object
phone                 object
mon_hrs               object
tue_hrs               object
wed_hrs               object
thu_hrs               object
fri_hrs               object
sat_hrs               object
sun_hrs               object
drive-thru              bool
mobile-order            bool
dunkin-card             bool
kcup                    bool
curbside-pickup         bool
has-baskin-robbins      bool
dtype: object

In [15]:
df_new[df_new['address_line_2'].isna()]

row_id              city state             address_line_1  \
1          1       Albertville    AL        8171 US Highway 431   
2          2            Athens    AL       1700 US Highway 72 E   
3          3            Auburn    AL           171 N College St   
4          4            Auburn    AL          2049 S College St   
5          5          Bessemer    AL     2178 Eastern Valley Rd   
...      ...               ...   ...                        ...   
9242    9242            Weston    WI         4020 Schofield Ave   
9244    9244   Wisconsin Dells    WI  2415 Wisconsin Dells Pkwy   
9245    9245   Wisconsin Dells    WI             69 Commerce St   
9246    9246  Wisconsin Rapids    WI              2121 8th St S   
9247    9247     FE Warren AFB    WY            6305 Missile Dr   

     address_line_2           phone            mon_hrs            tue_hrs  \
1               NaN  (256) 869-6075  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   
2               NaN  (256) 960-9302  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
3               NaN  (334) 707-8552  6:00 AM - 3:00 PM  6:00 AM - 3:00 PM   
4               NaN  (334) 501-2233  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
5               NaN  (205) 425-1333  6:00 AM - 4:00 PM  6:00 AM - 4:00 PM   
...             ...             ...                ...                ...   
9242            NaN  (715) 409-1843  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
9244            NaN  (608) 253-5353  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   
9245            NaN  (608) 350-9029  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   
9246            NaN  (715) 204-5260  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
9247            NaN  (307) 637-2818  6:00 AM - 5:00 PM  6:00 AM - 5:00 PM   

                wed_hrs            thu_hrs             fri_hrs  \
1     5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 10:00 PM   
2     5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   
3     6:00 AM - 3:00 PM  6:00 AM - 3:00 PM   6:00 AM - 3:00 PM   
4     5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   
5     6:00 AM - 4:00 PM  6:00 AM - 4:00 PM   6:00 AM - 4:00 PM   
...                 ...                ...                 ...   
9242  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   
9244  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   5:30 AM - 8:00 PM   
9245  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   5:30 AM - 8:00 PM   
9246  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   
9247  6:00 AM - 5:00 PM  6:00 AM - 3:00 PM   6:00 AM - 3:00 PM   

                 sat_hrs            sun_hrs  drive-thru  mobile-order  \
1     5:00 AM - 10:00 PM  5:00 AM - 9:00 PM        True          True   
2      5:00 AM - 8:00 PM  5:00 AM - 8:00 PM        True          True   
3      6:00 AM - 3:00 PM  6:00 AM - 3:00 PM       False         False   
4      5:00 AM - 8:00 PM  6:00 AM - 8:00 PM        True          True   
5      6:00 AM - 4:00 PM  7:00 AM - 4:00 PM        True          True   
...                  ...                ...         ...           ...   
9242   5:00 AM - 8:00 PM  5:00 AM - 8:00 PM        True          True   
9244   5:30 AM - 8:00 PM  5:30 AM - 8:00 PM        True          True   
9245   5:30 AM - 8:00 PM  5:30 AM - 8:00 PM        True          True   
9246   5:00 AM - 8:00 PM  5:00 AM - 8:00 PM        True         False   
9247   8:00 AM - 3:00 PM  8:00 AM - 3:00 PM        True          True   

      dunkin-card   kcup  curbside-pickup  has-baskin-robbins  
1            True  False            False                True  
2            True   True            False               False  
3            True   True            False               False  
4            True   True            False               False  
5            True   True            False               False  
...           ...    ...              ...                 ...  
9242         True   True            False               False  
9244         True   True            False               False  
9245         True   True      

In [16]:
df_new.drop('address_line_2', inplace=True, axis=1)
df_new.head(2)

row_id         city state       address_line_1           phone  \
0       0    Alabaster    AL        1114 1st St N  (205) 624-4880   
1       1  Albertville    AL  8171 US Highway 431  (256) 869-6075   

             mon_hrs            tue_hrs            wed_hrs            thu_hrs  \
0  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
1  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   

              fri_hrs             sat_hrs            sun_hrs  drive-thru  \
0   5:00 AM - 8:00 PM   6:00 AM - 8:00 PM  6:00 AM - 8:00 PM        True   
1  5:00 AM - 10:00 PM  5:00 AM - 10:00 PM  5:00 AM - 9:00 PM        True   

   mobile-order  dunkin-card   kcup  curbside-pickup  has-baskin-robbins  
0         False         True   True            False               False  
1          True         True  False            False                True

In [17]:
df_new["phone"].fillna('0123456789', inplace = True)

In [18]:
df_new.isna().sum()

row_id                0
city                  0
state                 0
address_line_1        0
phone                 0
mon_hrs               0
tue_hrs               0
wed_hrs               0
thu_hrs               0
fri_hrs               0
sat_hrs               0
sun_hrs               0
drive-thru            0
mobile-order          0
dunkin-card           0
kcup                  0
curbside-pickup       0
has-baskin-robbins    0
dtype: int64

In [19]:
df_new.drop('phone', inplace=True, axis='columns')
df_new.head(2)

row_id         city state       address_line_1            mon_hrs  \
0       0    Alabaster    AL        1114 1st St N  5:00 AM - 8:00 PM   
1       1  Albertville    AL  8171 US Highway 431  5:00 AM - 9:00 PM   

             tue_hrs            wed_hrs            thu_hrs  \
0  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
1  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   

              fri_hrs             sat_hrs            sun_hrs  drive-thru  \
0   5:00 AM - 8:00 PM   6:00 AM - 8:00 PM  6:00 AM - 8:00 PM        True   
1  5:00 AM - 10:00 PM  5:00 AM - 10:00 PM  5:00 AM - 9:00 PM        True   

   mobile-order  dunkin-card   kcup  curbside-pickup  has-baskin-robbins  
0         False         True   True            False               False  
1          True         True  False            False                True

In [20]:
df_new=df_new.rename(columns={'address_line_1':'address'})
df_new.head()

row_id         city state               address            mon_hrs  \
0       0    Alabaster    AL         1114 1st St N  5:00 AM - 8:00 PM   
1       1  Albertville    AL   8171 US Highway 431  5:00 AM - 9:00 PM   
2       2       Athens    AL  1700 US Highway 72 E  5:00 AM - 8:00 PM   
3       3       Auburn    AL      171 N College St  6:00 AM - 3:00 PM   
4       4       Auburn    AL     2049 S College St  5:00 AM - 8:00 PM   

             tue_hrs            wed_hrs            thu_hrs  \
0  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
1  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   
2  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
3  6:00 AM - 3:00 PM  6:00 AM - 3:00 PM  6:00 AM - 3:00 PM   
4  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   

              fri_hrs             sat_hrs            sun_hrs  drive-thru  \
0   5:00 AM - 8:00 PM   6:00 AM - 8:00 PM  6:00 AM - 8:00 PM        True   
1  5:00 AM - 10:00 PM  5:00 AM - 10:00 PM  5:00 AM - 9:00 PM        True   
2   5:00 AM - 8:00 PM   5:00 AM - 8:00 PM  5:00 AM - 8:00 PM        True   
3   6:00 AM - 3:00 PM   6:00 AM - 3:00 PM  6:00 AM - 3:00 PM       False   
4   5:00 AM - 8:00 PM   5:00 AM - 8:00 PM  6:00 AM - 8:00 PM        True   

   mobile-order  dunkin-card   kcup  curbside-pickup  has-baskin-robbins  
0         False         True   True            False               False  
1          True         True  False            False                True  
2          True         True   True            False               False  
3         False         True   True            False               False  
4          True         True   True            False               False

In [21]:
print(df_new.duplicated())

0       False
1       False
2       False
3       False
4       False
        ...  
9243    False
9244    False
9245    False
9246    False
9247    False
Length: 9248, dtype: bool


In [22]:
df_new.replace({False:0,True:1},inplace=True)

df_new.head()

row_id         city state               address            mon_hrs  \
0       0    Alabaster    AL         1114 1st St N  5:00 AM - 8:00 PM   
1       1  Albertville    AL   8171 US Highway 431  5:00 AM - 9:00 PM   
2       2       Athens    AL  1700 US Highway 72 E  5:00 AM - 8:00 PM   
3       3       Auburn    AL      171 N College St  6:00 AM - 3:00 PM   
4       4       Auburn    AL     2049 S College St  5:00 AM - 8:00 PM   

             tue_hrs            wed_hrs            thu_hrs  \
0  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
1  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   
2  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
3  6:00 AM - 3:00 PM  6:00 AM - 3:00 PM  6:00 AM - 3:00 PM   
4  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   

              fri_hrs             sat_hrs            sun_hrs  drive-thru  \
0   5:00 AM - 8:00 PM   6:00 AM - 8:00 PM  6:00 AM - 8:00 PM           1   
1  5:00 AM - 10:00 PM  5:00 AM - 10:00 PM  5:00 AM - 9:00 PM           1   
2   5:00 AM - 8:00 PM   5:00 AM - 8:00 PM  5:00 AM - 8:00 PM           1   
3   6:00 AM - 3:00 PM   6:00 AM - 3:00 PM  6:00 AM - 3:00 PM           0   
4   5:00 AM - 8:00 PM   5:00 AM - 8:00 PM  6:00 AM - 8:00 PM           1   

   mobile-order  dunkin-card  kcup  curbside-pickup  has-baskin-robbins  
0             0            1     1                0                   0  
1             1            1     0                0                   1  
2             1            1     1                0                   0  
3             0            1     1                0                   0  
4             1            1     1                0                   0

In [23]:
# split column and add new columns to df_new
df_new[['mon_opening_hrs', 'mon_closing_hrs']] = df_new['mon_hrs'].str.split('-', expand=True)
df_new[['tue_opening_hrs', 'tue_closing_hrs']] = df_new['tue_hrs'].str.split('-', expand=True)
df_new[['wed_opening_hrs', 'wed_closing_hrs']] = df_new['wed_hrs'].str.split('-', expand=True)
df_new[['thu_opening_hrs', 'thu_closing_hrs']] = df_new['thu_hrs'].str.split('-', expand=True)
df_new[['fri_opening_hrs', 'fri_closing_hrs']] = df_new['fri_hrs'].str.split('-', expand=True)
df_new[['sat_opening_hrs', 'sat_closing_hrs']] = df_new['sat_hrs'].str.split('-', expand=True)
df_new[['sun_opening_hrs', 'sun_closing_hrs']] = df_new['sun_hrs'].str.split('-', expand=True)

# display the dataframe
df_new

row_id              city state                    address  \
0          0         Alabaster    AL              1114 1st St N   
1          1       Albertville    AL        8171 US Highway 431   
2          2            Athens    AL       1700 US Highway 72 E   
3          3            Auburn    AL           171 N College St   
4          4            Auburn    AL          2049 S College St   
...      ...               ...   ...                        ...   
9243    9243     Whitefish Bay    WI     106 W Silver Spring Dr   
9244    9244   Wisconsin Dells    WI  2415 Wisconsin Dells Pkwy   
9245    9245   Wisconsin Dells    WI             69 Commerce St   
9246    9246  Wisconsin Rapids    WI              2121 8th St S   
9247    9247     FE Warren AFB    WY            6305 Missile Dr   

                mon_hrs            tue_hrs            wed_hrs  \
0     5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
1     5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   
2     5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
3     6:00 AM - 3:00 PM  6:00 AM - 3:00 PM  6:00 AM - 3:00 PM   
4     5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
...                 ...                ...                ...   
9243  6:00 AM - 6:00 PM  6:00 AM - 6:00 PM  6:00 AM - 6:00 PM   
9244  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   
9245  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   
9246  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
9247  6:00 AM - 5:00 PM  6:00 AM - 5:00 PM  6:00 AM - 5:00 PM   

                thu_hrs             fri_hrs             sat_hrs  ...  \
0     5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   6:00 AM - 8:00 PM  ...   
1     5:00 AM - 9:00 PM  5:00 AM - 10:00 PM  5:00 AM - 10:00 PM  ...   
2     5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   5:00 AM - 8:00 PM  ...   
3     6:00 AM - 3:00 PM   6:00 AM - 3:00 PM   6:00 AM - 3:00 PM  ...   
4     5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   5:00 AM - 8:00 PM  ...   
...                 ...                 ...                 ...  ...   
9243  6:00 AM - 6:00 PM   6:00 AM - 6:00 PM   6:00 AM - 6:00 PM  ...   
9244  5:30 AM - 8:00 PM   5:30 AM - 8:00 PM   5:30 AM - 8:00 PM  ...   
9245  5:30 AM - 8:00 PM   5:30 AM - 8:00 PM   5:30 AM - 8:00 PM  ...   
9246  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   5:00 AM - 8:00 PM  ...   
9247  6:00 AM - 3:00 PM   6:00 AM - 3:00 PM   8:00 AM - 3:00 PM  ...   

     wed_opening_hrs  wed_closing_hrs  thu_opening_hrs  thu_closing_hrs  \
0           5:00 AM           8:00 PM         5:00 AM           8:00 PM   
1           5:00 AM           9:00 PM         5:00 AM           9:00 PM   
2           5:00 AM           8:00 PM         5:00 AM           8:00 PM   
3           6:00 AM           3:00 PM         6:00 AM           3:00 PM   
4           5:00 AM           8:00 PM         5:00 AM           8:00 PM   
...              ...              ...              ...              ...   
9243        6:00 AM           6:00 PM         6:00 AM           6:00 PM   
9244        5:30 AM           8:00 PM         5:30 AM           8:00 PM   
9245        5:30 AM           8:00 PM         5:30 AM           8:00 PM   
9246        5:00 AM           8:00 PM         5:00 AM           8:00 PM   
9247        6:00 AM           5:00 PM         6:00 AM           3:00 PM   

      fri_opening_hrs  fri_closing_hrs  sat_opening_hrs sat_closing_hrs  \
0            5:00 AM           8:00 PM         6:00 AM          8:00 PM   
1            5:00 AM          10:00 PM         5:00 AM         10:00 PM   
2            5:00 AM           8:00 PM         5:00 AM          8:00 PM   
3            6:00 AM           3:00 PM         6:00 AM          3:00 PM   
4            5:00 AM           8:00 PM         5:00 AM          8:00 PM   
...               ...              ...              ...             ...   
9243         6:00 AM           6:00 PM         6:00 AM          6:00 PM   
9244         5:30 AM           8:00 PM         5:30 AM

In [24]:
df_new.columns

Index(['row_id', 'city', 'state', 'address', 'mon_hrs', 'tue_hrs', 'wed_hrs',
       'thu_hrs', 'fri_hrs', 'sat_hrs', 'sun_hrs', 'drive-thru',
       'mobile-order', 'dunkin-card', 'kcup', 'curbside-pickup',
       'has-baskin-robbins', 'mon_opening_hrs', 'mon_closing_hrs',
       'tue_opening_hrs', 'tue_closing_hrs', 'wed_opening_hrs',
       'wed_closing_hrs', 'thu_opening_hrs', 'thu_closing_hrs',
       'fri_opening_hrs', 'fri_closing_hrs', 'sat_opening_hrs',
       'sat_closing_hrs', 'sun_opening_hrs', 'sun_closing_hrs'],
      dtype='object')

In [25]:
dunkin_dataset=df_new[['row_id','city', 'state','address',
        'mon_opening_hrs', 'mon_closing_hrs', 'tue_opening_hrs',
       'tue_closing_hrs', 'wed_opening_hrs', 'wed_closing_hrs',
       'thu_opening_hrs', 'thu_closing_hrs', 'fri_opening_hrs',
       'fri_closing_hrs', 'sat_opening_hrs', 'sat_closing_hrs',
       'sun_opening_hrs', 'sun_closing_hrs', 'drive-thru', 'mobile-order', 'dunkin-card',
       'kcup', 'curbside-pickup', 'has-baskin-robbins']]
pd.set_option('display.max_columns', None)
dunkin_dataset.head()       


row_id         city state               address mon_opening_hrs  \
0       0    Alabaster    AL         1114 1st St N        5:00 AM    
1       1  Albertville    AL   8171 US Highway 431        5:00 AM    
2       2       Athens    AL  1700 US Highway 72 E        5:00 AM    
3       3       Auburn    AL      171 N College St        6:00 AM    
4       4       Auburn    AL     2049 S College St        5:00 AM    

  mon_closing_hrs tue_opening_hrs tue_closing_hrs wed_opening_hrs  \
0         8:00 PM        5:00 AM          8:00 PM        5:00 AM    
1         9:00 PM        5:00 AM          9:00 PM        5:00 AM    
2         8:00 PM        5:00 AM          8:00 PM        5:00 AM    
3         3:00 PM        6:00 AM          3:00 PM        6:00 AM    
4         8:00 PM        5:00 AM          8:00 PM        5:00 AM    

  wed_closing_hrs thu_opening_hrs thu_closing_hrs fri_opening_hrs  \
0         8:00 PM        5:00 AM          8:00 PM        5:00 AM    
1         9:00 PM        5:00 AM          9:00 PM        5:00 AM    
2         8:00 PM        5:00 AM          8:00 PM        5:00 AM    
3         3:00 PM        6:00 AM          3:00 PM        6:00 AM    
4         8:00 PM        5:00 AM          8:00 PM        5:00 AM    

  fri_closing_hrs sat_opening_hrs sat_closing_hrs sun_opening_hrs  \
0         8:00 PM        6:00 AM          8:00 PM        6:00 AM    
1        10:00 PM        5:00 AM         10:00 PM        5:00 AM    
2         8:00 PM        5:00 AM          8:00 PM        5:00 AM    
3         3:00 PM        6:00 AM          3:00 PM        6:00 AM    
4         8:00 PM        5:00 AM          8:00 PM        6:00 AM    

  sun_closing_hrs  drive-thru  mobile-order  dunkin-card  kcup  \
0         8:00 PM           1             0            1     1   
1         9:00 PM           1             1            1     0   
2         8:00 PM           1             1            1     1   
3         3:00 PM           0             0            1     1   
4         8:00 PM           1             1            1     1   

   curbside-pickup  has-baskin-robbins  
0                0                   0  
1                0                   1  
2                0                   0  
3                0                   0  
4                0                   0

In [26]:
final_data_set = dunkin_dataset[['row_id','city', 'state','address',
        'drive-thru', 'mobile-order', 'dunkin-card',
       'kcup', 'curbside-pickup', 'has-baskin-robbins']]

final_data_set.head()       

row_id         city state               address  drive-thru  mobile-order  \
0       0    Alabaster    AL         1114 1st St N           1             0   
1       1  Albertville    AL   8171 US Highway 431           1             1   
2       2       Athens    AL  1700 US Highway 72 E           1             1   
3       3       Auburn    AL      171 N College St           0             0   
4       4       Auburn    AL     2049 S College St           1             1   

   dunkin-card  kcup  curbside-pickup  has-baskin-robbins  
0            1     1                0                   0  
1            1     0                0                   1  
2            1     1                0                   0  
3            1     1                0                   0  
4            1     1                0                   0

In [27]:
item_counts=final_data_set['state'].value_counts()
print(item_counts)

NY    1419
MA    1075
FL     871
NJ     864
IL     697
PA     614
CT     486
MD     300
GA     252
OH     232
VA     226
NH     218
NC     195
TX     182
RI     163
ME     159
CA     127
TN     115
AZ     101
SC     100
IN      96
MI      96
WI      79
DE      66
AL      54
MO      53
VT      49
CO      44
KY      38
NV      36
IA      35
MN      32
KS      30
WV      26
NE      23
DC      21
NM      17
OK      14
LA      14
MS      10
AR       9
HI       6
AK       2
UT       1
WY       1
Name: state, dtype: int64


In [28]:
clean_states={'state':{'NY':1,"MA":2,'FL':3,"NJ":4,'IL':5,'PA':6,'CT':7,'MD':8,'GA':9,'OH':9,'VA':10,'NH':11 ,'NC':12,'TX':13,'RI':14,'ME':15,'CA':16,'TN':17,'AZ':18,'SC':19,
                        'IN':20,"MI":21,'WI':22,'DE':23,'AL':24,'MO':25,'VT':26,'CO':27,'KY':28,'NV':29,'IA':30,'MN':31,'KS':32,"WV":33,'NE':34,'DC':35,'NM':36,'OK':37,'LA':38,
                        'MS':41,"AR":42,"HI":43,'AK':44,'UT':45,'WY':46}}

In [29]:
final_data_set=final_data_set.replace(clean_states)
final_data_set.head()

row_id         city  state               address  drive-thru  mobile-order  \
0       0    Alabaster     24         1114 1st St N           1             0   
1       1  Albertville     24   8171 US Highway 431           1             1   
2       2       Athens     24  1700 US Highway 72 E           1             1   
3       3       Auburn     24      171 N College St           0             0   
4       4       Auburn     24     2049 S College St           1             1   

   dunkin-card  kcup  curbside-pickup  has-baskin-robbins  
0            1     1                0                   0  
1            1     0                0                   1  
2            1     1                0                   0  
3            1     1                0                   0  
4            1     1                0                   0

In [32]:
item_counts=final_data_set['address'].value_counts()
print(item_counts)

JFK Intl Airport         14
Logan Airport             8
Newark Int'L Airport      6
Ronald Reagan Airport     6
Dfw Int. Airport          5
                         ..
325 Gorham Rd             1
169 Main St               1
190 Route 3               1
175 Main St               1
6305 Missile Dr           1
Name: address, Length: 9144, dtype: int64


In [31]:
final_data_set.dtypes

row_id                 int64
city                  object
state                  int64
address               object
drive-thru             int64
mobile-order           int64
dunkin-card            int64
kcup                   int64
curbside-pickup        int64
has-baskin-robbins     int64
dtype: object

In [40]:
final_data_set.drop('address', inplace=True, axis='columns')
final_data_set.head()

row_id         city  state  drive-thru  mobile-order  dunkin-card  kcup  \
0       0    Alabaster     24           1             0            1     1   
1       1  Albertville     24           1             1            1     0   
2       2       Athens     24           1             1            1     1   
3       3       Auburn     24           0             0            1     1   
4       4       Auburn     24           1             1            1     1   

   curbside-pickup  has-baskin-robbins  
0                0                   0  
1                0                   1  
2                0                   0  
3                0                   0  
4                0                   0

In [42]:
df

row_id             address_line_1 address_line_2              city  \
0          0              1114 1st St N      Suite 200         Alabaster   
1          1        8171 US Highway 431            NaN       Albertville   
2          2       1700 US Highway 72 E            NaN            Athens   
3          3           171 N College St            NaN            Auburn   
4          4          2049 S College St            NaN            Auburn   
...      ...                        ...            ...               ...   
9243    9243     106 W Silver Spring Dr          Ste A     Whitefish Bay   
9244    9244  2415 Wisconsin Dells Pkwy            NaN   Wisconsin Dells   
9245    9245             69 Commerce St            NaN   Wisconsin Dells   
9246    9246              2121 8th St S            NaN  Wisconsin Rapids   
9247    9247            6305 Missile Dr            NaN     FE Warren AFB   

     state    zip           phone            mon_hrs            tue_hrs  \
0       AL  35007  (205) 624-4880  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
1       AL  35950  (256) 869-6075  5:00 AM - 9:00 PM  5:00 AM - 9:00 PM   
2       AL  35611  (256) 960-9302  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
3       AL  36830  (334) 707-8552  6:00 AM - 3:00 PM  6:00 AM - 3:00 PM   
4       AL  36832  (334) 501-2233  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
...    ...    ...             ...                ...                ...   
9243    WI  53217  (414) 284-7608  6:00 AM - 6:00 PM  6:00 AM - 6:00 PM   
9244    WI  53965  (608) 253-5353  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   
9245    WI  53965  (608) 350-9029  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   
9246    WI  54494  (715) 204-5260  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   
9247    WY  82005  (307) 637-2818  6:00 AM - 5:00 PM  6:00 AM - 5:00 PM   

                wed_hrs            thu_hrs             fri_hrs  \
0     5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   
1     5:00 AM - 9:00 PM  5:00 AM - 9:00 PM  5:00 AM - 10:00 PM   
2     5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   
3     6:00 AM - 3:00 PM  6:00 AM - 3:00 PM   6:00 AM - 3:00 PM   
4     5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   
...                 ...                ...                 ...   
9243  6:00 AM - 6:00 PM  6:00 AM - 6:00 PM   6:00 AM - 6:00 PM   
9244  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   5:30 AM - 8:00 PM   
9245  5:30 AM - 8:00 PM  5:30 AM - 8:00 PM   5:30 AM - 8:00 PM   
9246  5:00 AM - 8:00 PM  5:00 AM - 8:00 PM   5:00 AM - 8:00 PM   
9247  6:00 AM - 5:00 PM  6:00 AM - 3:00 PM   6:00 AM - 3:00 PM   

                 sat_hrs            sun_hrs  drive-thru  mobile-order  \
0      6:00 AM - 8:00 PM  6:00 AM - 8:00 PM        True         False   
1     5:00 AM - 10:00 PM  5:00 AM - 9:00 PM        True          True   
2      5:00 AM - 8:00 PM  5:00 AM - 8:00 PM        True          True   
3      6:00 AM - 3:00 PM  6:00 AM - 3:00 PM       False         False   
4      5:00 AM - 8:00 PM  6:00 AM - 8:00 PM        True          True   
...                  ...                ...         ...           ...   
9243   6:00 AM - 6:00 PM  6:00 AM - 4:00 PM       False         False   
9244   5:30 AM - 8:00 PM  5:30 AM - 8:00 PM        True          True   
9245   5:30 AM - 8:00 PM  5:30 AM - 8:00 PM        True          True   
9246   5:00 AM - 8:00 PM  5:00 AM - 8:00 PM        True         False   
9247   8:00 AM - 3:00 PM  8:00 AM - 3:00 PM        True          True   

      dunkin-card   kcup  curbside-pickup  has-baskin-robbins    loc_lat  \
0            True   True            False               False        NaN   
1            True  False            False                True        NaN   
2            True   True            False               False  34.781698   
3            True   True            False               False  32.608800   
4            True   True            False               False  32.567323   
...           ...    ...              ...                 ...     

In [43]:
final_data_set

row_id              city  state  drive-thru  mobile-order  dunkin-card  \
0          0         Alabaster     24           1             0            1   
1          1       Albertville     24           1             1            1   
2          2            Athens     24           1             1            1   
3          3            Auburn     24           0             0            1   
4          4            Auburn     24           1             1            1   
...      ...               ...    ...         ...           ...          ...   
9243    9243     Whitefish Bay     22           0             0            1   
9244    9244   Wisconsin Dells     22           1             1            1   
9245    9245   Wisconsin Dells     22           1             1            1   
9246    9246  Wisconsin Rapids     22           1             0            1   
9247    9247     FE Warren AFB     46           1             1            1   

      kcup  curbside-pickup  has-baskin-robbins  
0        1                0                   0  
1        0                0                   1  
2        1                0                   0  
3        1                0                   0  
4        1                0                   0  
...    ...              ...                 ...  
9243     1                0                   0  
9244     1                0                   0  
9245     1                0                   0  
9246     1                0                   0  
9247     1                0                   1  

[9248 rows x 9 columns]

In [44]:
final_data_set.to_csv('dunkin_locations.csv',index=False)